### PR TITLE
Add clarity to density test

### PR DESCRIPTION
**Pull request by bprashanth**
https://github.com/GoogleCloudPlatform/kubernetes/issues/10282

### CODE REVIEW

**Comment by wojtek-t** [comment]
For the reference see also #10345 - the explanation from there is related to this one.

**Comment by mbforbes** [comment]
can't merge--rebase please?

**Comment by bprashanth** [comment]
@quinton-hoole lets wait for https://github.com/GoogleCloudPlatform/kubernetes/pull/10313 to avoid another merge conflict (since you're the reviewer of both)

File: test/e2e/kubelet_stats.go [hunk 1]
```diff
@@ -159,17 +160,30 @@ func HighLatencyKubeletOperations(c *client.Client, threshold time.Duration, nod
 	return badMetrics, nil
 }
 
-// Retrieve metrics from the kubelet server of the given node.
-func getKubeletMetrics(c *client.Client, node string) (string, error) {
-	metric, err := c.Get().
+// Performs a get on a node proxy endpoing given the nodename and rest client.
+func nodeProxyRequest(c *client.Client, node, endpoint string) client.Result {
```
Comment (quinton-hoole) [inline test/e2e/kubelet_stats.go]: nit: node doesn't seem like the right variable name here. nodeName?  I don't feel terribly strongly about this one.

File: test/e2e/kubelet_stats.go [hunk 2]
```diff
@@ -159,17 +160,30 @@ func HighLatencyKubeletOperations(c *client.Client, threshold time.Duration, nod
...
-		Suffix("metrics").
-		Do().
-		Raw()
+		Suffix(endpoint).
+		Do()
+}
+
+// Retrieve metrics from the kubelet server of the given node.
+func getKubeletMetrics(c *client.Client, node string) (string, error) {
+	metric, err := nodeProxyRequest(c, node, "metrics").Raw()
```
Comment (quinton-hoole) [inline test/e2e/kubelet_stats.go]: As per previous comment.  nodeName?

File: test/e2e/util.go [hunk 3]
```diff
@@ -1032,6 +1032,11 @@ func RunRC(config RCConfig) error {
 		}
 		oldPods = pods
 		oldRunning = running
+
```
Comment (quinton-hoole) [inline test/e2e/util.go]: Just for the record, this function is now 128 lines long - over-ripe for a refactoring, but not necessarily in this PR.

**Comment by quinton-hoole** [comment]
minor optional nits in feedback.  LGTM. Will wait for #10313 to merge before marking this ok-to-merge.

**Comment by quinton-hoole** [comment]
Still blocked on @dchen1107 or @lavalamp to review a small amount of kubelet code in #10313

**Comment by bprashanth** [comment]
Dependent pr is in, I've reabased, this only modifies a test so it should be low risk.

**Comment by bprashanth** [comment]
@k8s-bot ok to test

**Comment by quinton-hoole** [comment]
ok-to-merge

**Pull request merged**